crates/rpc/rpc/src/otterscan.rs

```
@@ -0,0 +1,85 @@
+#![allow(dead_code, unused_variables)]
+use async_trait::async_trait;
+use jsonrpsee::core::RpcResult;
+use reth_primitives::{Address, BlockId, TxHash, H256, U256};
+use reth_rpc_api::{EthApiServer, OtterscanServer};
+use reth_rpc_types::{
+    BlockDetails, ContractCreator, InternalOperation, TraceEntry, Transaction,
+    TransactionsWithReceipts,
+};
+
+/// Otterscan Api
+#[derive(Debug)]
+pub struct OtterscanApi<Eth> {
+    eth: Eth,
+}
+
+impl<Eth> OtterscanApi<Eth> {
+    /// Creates a new instance of `Otterscan`.
+    pub fn new(eth: Eth) -> Self {
+        Self { eth }
+    }
+}
+#[async_trait]
+impl<Eth> OtterscanServer for OtterscanApi<Eth>
+where
+    Eth: EthApiServer,
+{
+    async fn has_code(&self, address: Address, block_number: Option<BlockId>) -> RpcResult<bool> {
+        todo!()
```

we want to return an rpc error here instead, like unimplemented, so we're not able to cause a panic